edit/edit.go

```diff
@@ -136,21 +136,34 @@ func InterpretLabelForWorkspaceLocation(root string, target string) (buildFile s
 	}
 
 	if strings.HasPrefix(target, "//") {
-		buildFile = path.Join(rootDir, pkg, "BUILD")
+		buildFile = filepath.Join(rootDir, pkg, "BUILD")
+		if !isFile(buildFile) {
+			// try it with the .bazel extension
+			buildFile += ".bazel"
+		}
 		return
 	}
 	if isFile(pkg) {
 		// allow operation on other files like WORKSPACE
 		buildFile = pkg
-		pkg = path.Join(relativePath, filepath.Dir(pkg))
+		pkg = filepath.Join(relativePath, filepath.Dir(pkg))
 		return
 	}
 	if pkg != "" {
 		buildFile = pkg + "/BUILD"
+		if !isFile(buildFile) {
+			// try it with the .bazel extension
+			buildFile += ".bazel"
+		}
+
 	} else {
 		buildFile = "BUILD"
+		if !isFile(buildFile) {
```

Comment:
Instead of duplicating the check, you can move it after the if-else block.

Reply:
pushed the change for that